integration_tests/suite/base/test_context_range.py

```diff
@@ -0,0 +1,290 @@
+        {'start': '1000', 'end': '1099'},
+        {'start': '1200', 'end': '1499'},
+        {'start': '1800', 'end': '1899'},
+    ],
+    group_ranges=[
+        {'start': '2000', 'end': '2999'},
+    ],
+    queue_ranges=[
+        {'start': '3000', 'end': '3999'},
+    ],
+    conference_room_ranges=[
+        {'start': '4000', 'end': '4999'},
+    ],
+)
+@fixtures.context(
+    type='incall',
+    incall_ranges=[
+        {'start': '8005551000', 'end': '8005551009'},
+    ],
+)
+def test_search_all_types(internal_context, incoming_context):
+    response = confd.contexts(internal_context['id']).ranges('user').get()
+    assert_that(
+        response.json,
+        has_entries(
+            total=3,
+            items=contains_inanyorder(
+                {'start': '1000', 'end': '1099'},
+                {'start': '1200', 'end': '1499'},
+                {'start': '1800', 'end': '1899'},
+            ),
+        ),
+    )
+
+    response = confd.contexts(internal_context['id']).ranges('group').get()
+    assert_that(
+        response.json,
+        has_entries(
+            total=1,
+            items=contains_inanyorder(
+                {'start': '2000', 'end': '2999'},
+            ),
+        ),
+    )
+
+    response = confd.contexts(internal_context['id']).ranges('queue').get()
+    assert_that(
+        response.json,
+        has_entries(
+            total=1,
+            items=contains_inanyorder(
+                {'start': '3000', 'end': '3999'},
+            ),
+        ),
+    )
+
+    response = confd.contexts(internal_context['id']).ranges('conference').get()
+    assert_that(
+        response.json,
+        has_entries(
+            total=1,
+            items=contains_inanyorder(
+                {'start': '4000', 'end': '4999'},
+            ),
+        ),
+    )
+
+    response = confd.contexts(incoming_context['id']).ranges('incall').get()
+    assert_that(
+        response.json,
+        has_entries(
+            total=1,
+            items=contains_inanyorder(
+                {'start': '8005551000', 'end': '8005551009'},
+            ),
+        ),
+    )
+
+
+@fixtures.context(
+    user_ranges=[
+        {'start': '1000', 'end': '1999'},
+    ],
+)
+@fixtures.user()
+@fixtures.line_sip()
+def test_search_available(context, user, line):
+    @fixtures.extension(context=context['name'], exten='1005')
+    def aux(extension):
+        with a.user_line(user, line), a.line_extension(line, extension):
+            response = confd.contexts(context['id']).ranges('user').get(available=True)
+            assert_that(
+                response.json,
+                has_entries(
+                    total=2,
+                    items=contains_inanyorder(
+                        {'start': '1000', 'end': '1004'},
+                        {'start': '1006', 'end': '1999'},
+                    ),
+                ),
+            )
+
+            response = confd.contexts(context['id']).ranges('user').get(available=False)
```

Comment:
IMHO, available=False may mean for people that we want only the extensions that are not available.

Maybe available_only / only_available instead ???

Reply:
You're right, Francis made a similar comment. I'll try to find something better.